presto-main/src/main/java/com/facebook/presto/spiller/AesSpillCipher.java

```diff
@@ -0,0 +1,190 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spiller;
+
+import com.facebook.presto.spi.PrestoException;
+
+import javax.crypto.Cipher;
+import javax.crypto.KeyGenerator;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.IvParameterSpec;
+
+import java.nio.ByteBuffer;
+import java.security.GeneralSecurityException;
+import java.security.NoSuchAlgorithmException;
+
+import static com.facebook.presto.spi.StandardErrorCode.GENERIC_INTERNAL_ERROR;
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static java.lang.String.format;
+
+final class AesSpillCipher
+        implements SpillCipher
+{
+    //  256-bit AES CBC mode
+    private static final String CIPHER_NAME = "AES/CBC/PKCS5Padding";
+    private static final int KEY_BITS = 256;
+
+    private SecretKey key;
+    private final int ivBytes;
+
+    AesSpillCipher()
+    {
+        this.key = generateNewSecretKey();
+        this.ivBytes = createEncryptCipher().getIV().length;
+    }
+
+    @Override
+    public ByteBuffer decrypt(ByteBuffer encryptedData)
+    {
+        int length = encryptedData.remaining();
+        if (length < ivBytes) {
+            throw new PrestoException(GENERIC_INTERNAL_ERROR, format("Encrypted data size is too small: %s. It must be at least the size of the initialization vector: %s.", length, ivBytes));
+        }
+        byte[] iv = new byte[ivBytes];
+        encryptedData.get(iv);
+        Cipher cipher = createDecryptCipher(new IvParameterSpec(iv));
+        ByteBuffer output = ByteBuffer.allocate(cipher.getOutputSize(encryptedData.remaining()));
+        try {
+            cipher.doFinal(encryptedData, output);
+            output.flip();
+            return output;
+        }
+        catch (GeneralSecurityException e) {
+            throw new PrestoException(GENERIC_INTERNAL_ERROR, "Cannot decrypt previously encrypted data: " + e.getMessage(), e);
+        }
+    }
+
+    @Override
+    public byte[] decrypt(byte[] encryptedData)
+    {
+        return unwrapOrCopyToByteArray(decrypt(ByteBuffer.wrap(encryptedData)));
+    }
+
+    @Override
+    public ByteBuffer encrypt(ByteBuffer data)
+    {
+        Cipher cipher = createEncryptCipher();
+        byte[] iv = cipher.getIV();
+        ByteBuffer output = ByteBuffer
+                .allocate(iv.length + cipher.getOutputSize(data.remaining()))
+                .put(iv);
+        try {
+            cipher.doFinal(data, output);
+            output.flip();
+            return output;
+        }
+        catch (GeneralSecurityException e) {
+            throw new PrestoException(GENERIC_INTERNAL_ERROR, "Failed to encrypt data: " + e.getMessage(), e);
+        }
+    }
+
+    @Override
+    public byte[] encrypt(byte[] data)
+    {
+        return unwrapOrCopyToByteArray(encrypt(ByteBuffer.wrap(data)));
+    }
+
+    @Override
+    public void destroy()
+    {
+        this.key = null;
```

Comment:
Add a comment explaining why this is useful. On the surface, it doesn't seem necessary.

Reply:
Will add a comment to the method.